### PR TITLE
Redirect path to next path when user logs in 

### LINKED ISSUE #4112: Use redirect path if available when logging in and already logged in
### Describe the problem and steps to reproduce it:
When you are already logged in to AMO and try logging in again you get an error telling you that you are already logged in. This error is shown at /users/login/ and should instead redirect the user to the `to` path if it is valid, otherwise /users/login/ is fine.

1. Open AMO in two tabs.
2. Log in in one tab.
3. Log in in the other tab.

### What happened?
You are redirected to /users/login/ and shown an error.

### What did you expect to happen?
You are redirect to wherever you clicked the "Log in" button and shown an error.

### Anything else we should know?
ValentinaPC's screencast

From #4070.

### PR DESCRIPTION
**Pull request by vijeth-aradhya**
Fixed #4112

If the user is already logged in, and tries to log in, then redirect to next path instead of error on user login

### CODE REVIEW

**Comment by vijeth-aradhya** [comment]
@mstriemer The checks are showing wrong in tests that I updated. This is expected right?

File: src/olympia/devhub/templates/devhub/addons/listing/delete_form.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
   {% endif %}
   <p>
     <label>
-      {{ _('Enter the following text confirm your decision: {slug}')|fe(slug=addon.slug) }}
+      {{ _('Enter the following text to confirm your decision: {slug}')|fe(slug=addon.slug) }}
```
Comment (mstriemer) [inline src/olympia/devhub/templates/devhub/addons/listing/delete_form.html]: I think this was included accidentally, can you remove it from this pull request, please?

It looks like you're working on the master branch. I'd suggest you work on a new branch for each change.

**Comment by mstriemer** [comment]
I think CircleCI is failing for something unrelated today. I suggested a fix for the other test failure above.

GitHub has some [documentation](https://guides.github.com/introduction/flow/) on the sort of workflow I'm suggesting. You'll need to get your master branch back to matching mozilla/addons-server first though, might want to wait until we get this code landed to do that.

**Comment by vijeth-aradhya** [comment]
I have made the requested changes.
Um do you think I rebase and match with the current repo as well?

**Comment by mstriemer** [comment]
I'll squash the commits when I merge this. I think there's a fix landed for the CircleCI failure so I'm going to go ahead and merge this. It should pass on master.

**Comment by mstriemer** [comment]
Thanks for your work on this!

**Pull request merged**